### ISSUE TITLE
Setting bundleDepth to 1 causes steal-tools to throw an execption

### ISSUE THREAD

**Issue by RALifeCoach** [stealjs/steal-tools#520]
Using https://github.com/RALifeCoach/canjsbuildit
set bundleDepth to 1 and then run the build. The following exception is thrown. I want to be able to specify depth of 1 to minimize file loading at run time.

OPENING: buildit/app
OPENING: buildit@1.0.0#dev1/main1
OPENING: buildit@1.0.0#dev2/main2
Transpiling...
Minifying...
Calculating main bundle(s)...
Flattening main bundle(s)...
Calculating progressively loaded bundle(s)...
Flattening progressively loaded bundle(s)...
Potentially unhandled rejection [194] TypeError: Cannot read property 'bundles' of undefined
    at getName (/git/canjsbuiltit/node_modules/steal-tools/lib/bundle/name.js:163:27)
    at makeGetName (/git/canjsbuiltit/node_modules/steal-tools/lib/bundle/name.js:112:10)
    at Function.exports.getName (/git/canjsbuiltit/node_modules/steal-tools/lib/bundle/name.js:243:9)
    at Object.bundle.merge (/git/canjsbuiltit/node_modules/steal-tools/lib/bundle/flatten.js:35:39)
    at Object.bundle.flatten (/git/canjsbuiltit/node_modules/steal-tools/lib/bundle/flatten.js:16:12)
    at module.exports (/git/canjsbuiltit/node_modules/steal-tools/lib/bundle/flatten.js:155:9)
    at bundle (/git/canjsbuiltit/node_modules/steal-tools/lib/stream/bundle.js:79:3)
    at DestroyableTransform._transform (/git/canjsbuiltit/node_modules/steal-tools/lib/stream/bundle.js:24:17)
    at DestroyableTransform.Transform._read (/git/canjsbuiltit/node_modules/steal-tools/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:159:10)
    at DestroyableTransform.Transform._write (/git/canjsbuiltit/node_modules/steal-tools/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:147:83)

**Comment by matthewp** [comment]
Can confirm the bug.  Two problems:

1. As described, if you set bundleDepth to 1 you get this error.
2. bundleDepth is expected to be in the first argument SystemConfig when it should be in the second argument, BuildOptions.

**Comment by matthewp** [comment]
Have a PR in. Included is moving the bundleDepth option to where it should be:

```js
stealTools.build({
  config: __dirname + "/package.json!npm"
}, {
  bundleDepth: 1
});
```

**Comment by matthewp** [comment]
This was released in a 0.16 patch and I'm working on getting it into 1.0 now.

**Comment by matthewp** [comment]
Also released in 1.0.0-rc.1
